### ISSUE TITLE
Base path config option is not included in REST API responses when using non-default domains

### ISSUE THREAD

**Issue by sgohl** [shlinkio/shlink#990]
I recently opened a feature request https://github.com/shlinkio/shlink/issues/989

The answer based on the assumption of serving the app on the `BASE_PATH`, which isn't actually the problem really.

I have **NOT** set `SHORT_DOMAIN_HOST` anymore on purpose (which didn't make a difference anyway).

I need multiple domains and a fixed suffix prepended before the `short_code` on **generating the shortcode**
This is not related to serving the server-side app in a sub-URI-path.

example:

```
curl 'https://shlink.example.com/s/rest/v2/short-urls' ... --data-raw '{"longUrl":"https://my-long.com/URL","domain":"myprimeportal.com","findIfExists":true,"validateUrl":false}'

{"shortCode":"mKM9r","shortUrl":"https://myprimeportal.com/mKM9r" ...
```

the actual generated short-URL is 

https://myprimeportal.com/mKM9r

instead of what I want:

https://myprimeportal.com/s/mKM9r

Error:

```
curl 'https://shlink.example.com/s/rest/v2/short-urls' -H 'Content-Type: application/json;charset=utf-8' -H "X-Api-Key: $KEY" --data-raw '{"longUrl":"https://myprimeportal.com/Long/URL/with?parameters=true","tags":["something"],"domain":"myprimeportal.com/s","findIfExists":true,"validateUrl":false}'

{"invalidElements":["domain"],"title":"Invalid data","type":"INVALID_ARGUMENT","status":400,"detail":"Provided data is not valid"}~ > 
```
It does not accept `/s` within domain key. But giving the default domain exactly this way with appended `/s` , it works.

if I edit mysql table `short_urls` and prepend `s/` to field `short_code`, the app will list the wanted short-URL in the web-dashboard

This (appended `BASE_PATH`) only works with single domain, but not if you send a custom domain in the payload, then the `s/` is missing from the generated shortURL response.

any ideas ?

(using the docker `shlink:stable` behind traefik:v1 with `PathPrefix:/s/` and `BASE_PATH=/s`)

EDIT: `BASE_PATH` should be appended to all domains, not only the default domain; OR; make possible to payload a domain with URI path

**Comment by acelaya** [comment]
Thanks for confirming. It's a smaller issue then.

**Comment by sgohl** [comment]
nice, i really appreciate your investigation! if the listing would also include the BASE_PATH for every domain, I'd be very happy

**Comment by acelaya** [comment]
Yep, definitely, that needs to be fixed. I just wanted to confirm exactly what was the issue.

**Comment by acelaya** [comment]
Hey @sgohl. I have been testing this and I can't manage to reproduce it. For me it is working as expected.

If I provide the `BASE_PATH` to the docker image, the generated short URL in the response includes it.

I have followed these steps:

* Spin-up the docker image, including the base path:

    ```bash
    docker run --rm --name my_shlink -p 8081:8080 -e SHORT_DOMAIN_HOST=localhost:8081 -e SHORT_DOMAIN_SCHEMA=http -e BASE_PATH=\/s shlinkio/shlink:stable
    ```

* On a different terminal, generate an API key over the running container:

    ```bash
    docker exec -it my_shlink shlink api-key:generate
    ```

* Copy the generated API key, then run this command to generate a short URL (notice you have to replace `[API_KEY]` with the value returned in previous step):

    ```bash
    curl -X POST "http://localhost:8081/s/rest/v2/short-urls" -H  "accept: application/json" -H  "X-Api-Key: [API_KEY]" -H  "Content-Type: application/json" -d "{\"longUrl\":\"https://github.com/shlinkio/shlink/issues/990\"}"
    ```

After this, the response includes the `shortUrl` property with the `/s` path, as expected.

![image](https://user-images.githubusercontent.com/2719332/106397757-46cc1900-640f-11eb-8d0c-8a3698969b52.png)

Could it be that you are doing something differently?

Maybe you are not properly providing the env var to the shlink container, but just to traefik, which in turn properly redirects the request to Shlink but stripping-off the path?

**Comment by acelaya** [comment]
I'm stupid. The issue was reproducible only when using custom domains. Ignore my  previous comment 😅

**Comment by acelaya** [comment]
I reproduced it and fixed it. It will work as expected in v2.6.0

**Comment by sgohl** [comment]
thanks so much for your efforts, really appreciate!

I'll test and verify as soon as I find a 2.6.0 docker image

**Comment by acelaya** [comment]
Hey @sgohl, I released v2.6.0 yesterday, which includes the fix for this.

Give it a try and let me know if there's anything not working as expected.
